.github/workflows/ci.yaml

```diff
@@ -177,6 +177,15 @@ jobs:
           name: npm-package
           path: ./package.tar.gz
 
+      - name: Publish npm package with dev tag
+        run: yarn publish:npm:dev
```

Comment:
❓The job this runs in already has the npm artifact. Should we modify the script to use that or should we keep as is? The benefit of refactoring would be to speed things up. The benefit of keeping the current approach is this step doesn't rely on any side effects and is _pure_.

Reply:
My first instinct is to use the artifact on disk.  For example if I ran this locally for some reason I think it would be unexpected if it pulled from CI instead?  

Network access I would say is especially impure so I might argue downloading is less pure than using the artifact that was just built in the job.

Lastly the download script has always felt to me like a janky workaround we have for passing artifacts around jobs; when/if we automate the release I think it is possible it ends up going away.

Reply:
But if we did have to separate it into a job we could just do that easily with the current setup, so I see what you mean by being pure.

Reply:
Hm I think you are right to have it this way, moving it into another job seems like a definite possibility to me.  In fact maybe we should do that?  It might make more sense than to be part of the build job.

Reply:
> Lastly the download script has always felt to me like a janky workaround we have for passing artifacts around jobs; when/if we automate the release I think it is possible it ends up going away.

I agree. I had to dig in to remember how it works. I'd say let's use the artifact on disk like you suggested.

Reply:
Especially if we combine this with the existing NPM publish script because that one has to use the download as it is in a separate job.

I do still think we should replace the download script at some point, there has got to be a more official way.